### ISSUE TITLE
Benchmark memory and cpu processing for PixelSpotDecoder

### ISSUE THREAD

**Issue by ambrosejcarr** [spacetx/starfish#1283]
Multiple groups identified that the PixelSpotDecoder was too slow to run on 3d data, and that it takes too much RAM. 

One dataset required 60 GB to process a single (54, 1024, 1024) volume. 

Masking ineligible pixel, ask discussed by @kevinyamauchi and Dan Goodwin at the SpaceTx hackathon, could provide further mitigation, but are unlikely to provide an adequate speed-up alone.

**Comment by ambrosejcarr** [comment]
Seems the speed problem is the `sklearn.NearestNeighbors.kneighbors` call, or an interaction with multiprocessing (`parallel.py`)

![image](https://user-images.githubusercontent.com/5239152/57052292-6b9cc680-6c54-11e9-9092-aa0af0ed7168.png)

**Comment by shanaxel42** [comment]
repeat of #1312
